airflow/dag_processing/manager.py

```diff
@@ -599,7 +599,10 @@ def _run_parsing_loop(self):
 
             if self.standalone_dag_processor:
                 self._fetch_callbacks(max_callbacks_per_loop)
-            self._scan_stale_dags()
+
+            # this variable gives us flexibility to purge stale dags or not.
+            if not conf.get("core", "disable_scan_stale_dags", fallback=None):
```

Comment:
```suggestion
            if not conf.getboolean("core", "disable_scan_stale_dags", fallback=False):
```

We have a parser for the potential boolean, as well as better to provide default value explicitly